### DISCUSSION TITLE
How to use static-adapter with partial CSR for dynamic routes?

### DISCUSSION THREAD

**Discussion by HuakunShen** [sveltejs/kit#12936] [Q&A]
For my use case, I have to have a static site, so I believe the `static-adapter` is the only choice.
In the app `/store` lists a list of items fetched from database (dynamic), to will `goto()` `/store/[id]` when an item is selected. 
The app should fetch data from database and render the `[id]` page. These are all dynamic, and I don't expect them to be pre-renderred.

Another route is `/settings` and is expected to be pre-renderred. 

I couldn't figure out how to do this. 
Error I get looks like this
```
Error: The following routes were marked as prerenderable, but were not prerendered because they were not found while crawling your app:
  - /store/[id]
```

I've tried to set `export const csr=true` in `/store/[id]/+page.ts` and `/store/[id]/+layout.ts` but don't seem to work.
The only option to get the app built is by setting `fallback: "index.html"` in config, but this seems to turn the entire app into a CSR SPA and I don't want this.

Is there a way to pre-render some pages and make some pages CSR?

**Comment by brunnerh** [comment]
`fallback` usually should be `'404.html'`, which then should automatically be served if no pre-rendered route is found (at least most file servers support this sort of thing). And yes, those routes need CSR without pre-rendering if not all possible IDs are known and specified beforehand.

**Reply by HuakunShen** [reply]
So how to enable CSR for the dynamic pages? 
I couldn't figure out how to do CSR for `/store/[id]`

**Reply by brunnerh** [reply]
Add [a `+page.js`/`ts`](https://svelte.dev/docs/kit/page-options) to that directory and change `csr`/`prerender` accordingly.

**Reply by HuakunShen** [reply]
![image](https://github.com/user-attachments/assets/a9c129b9-651f-4f1b-a753-7f85e2f5d32c)
It doesn't seem to work for me.

Other than this file, the only place I have a render mode setting is the `+layout.ts` in root routes folder.

```ts
export const prerender = true
export const ssr = false
```

**Reply by brunnerh** [reply]
Did you set the `fallback` as well?

**Reply by HuakunShen** [reply]
No I didn't se `fallback`, should I set it to 'index.html`? I thought it will turn the entire app into CSR SPA (including other pages).

**Reply by brunnerh** [reply]
As far as I know , it should not do that, and as I said, it usually should be `'404.html'` so it is served for routes that are not found (because they did not get pre-rendered).

**Reply by HuakunShen** [reply]
Thank you I got it working!
I saw in the docs https://svelte.dev/docs/kit/adapter-static#Options-fallback that `index.html`, `404.html`, `200.html`

I tried them and they work the same way.
Thank you again!
